superset-frontend/src/dashboard/components/RefreshIntervalModal.tsx

```diff
@@ -91,6 +100,49 @@ class RefreshIntervalModal extends React.PureComponent<
     this.setState({
       refreshFrequency: value || refreshIntervalOptions[0][0],
     });
+
+    this.setState({
+      custom_block: value === -1,
+    });
+
+    if (value === -1) {
+      this.setState({
+        custom_hour: 0,
+        custom_min: 0,
+        custom_sec: 0,
+      });
+    }
+  }
+
+  onSaveValue(value: number) {
+    this.props.onChange(value, this.props.editMode);
+    this.modalRef?.current?.close();
+    this.props.addSuccessToast(t('Refresh interval saved'));
+  }
+
+  createIntervalOptions(refreshIntervalOptions: [number, string][]) {
+    const refresh_options = [];
+    if (!refreshIntervalOptions.length) {
+      refresh_options.push({ value: -1, label: 'Custom interval' });
+      return refresh_options;
+    }
+
+    refresh_options.push({ value: -1, label: 'Custom interval' });
```

Comment:
The label needs to be localized. Also, aren't you adding this two times as you are adding it when the above condition triggers too?

Reply:
Ok i will localize it,
no i am not adding twice because from if condition i am returning.

Reply:
What I mean is that you are duplicating the code by using the condition `!refreshIntervalOptions.length` and then pushing the custom interval option, but you also do the same when `refreshIntervalOptions.length` is larger than 0.

So just do:

```
refresh_options.push({ value: -1, label: 'Custom interval' });

if (refreshIntervalOptions.length) {
   refresh_options.push(
      ...refreshIntervalOptions.map(option => ({
        value: option[0],
        label: t(option[1]),
      })),
    );
}

return refresh_options;
```